src/api/ready.js

```diff
@@ -0,0 +1,54 @@
+import data from '../util/data';
+import registry from '../global/registry';
+
+function ready (element) {
+  const components = registry.find(element);
+  for (let a = 0; a < components.length; a++) {
+    if (!data(element, `lifecycle/${components[a].id}`).created) {
+      return false;
+    }
+  }
+  return true;
+}
+
+export default function (elements, callback) {
+  // If a selector is provided we convert it to to a collection of elements.
+  if (typeof elements === 'string') {
+    elements = document.querySelectorAll(elements);
+  }
+
+  // Ensure the collection is traversable.
+  const collection = elements.length === undefined ? [elements] : elements;
+  const collectionLength = collection.length;
+
+  // If there's no elements we don't do anything.
+  if (!collection.length) {
```

Comment:
add an node-style error parameter to the callback that you populate in that case? But I think the API should be stable, e.g. if there is a callback passed, it should be called at some point in time. For no elements (e.g. all of the 0 elements are ready :-)) I'd have expected this to be called straight away I think.